### PR TITLE
Initial dataset

### PR DESCRIPTION
**Pull request by fcpranav**
Initial dataset creation:
1. Generates roughly 2000 captcha images consisting of numerical characters only

Caveats:
1. The statistics printed at the end of the script are off right now, does not account for overwriting existing captcha images. This shouldn't really matter though, what is important is that they show that each character occurs more than once and on average, similar amounts of time.

image

### CODE REVIEW

**Comment by natanleung** [comment]
LGTM. But why are you committing any empty zip?

**Comment by natanleung** [comment]
> Initial dataset creation:
> 
> 1. Generates roughly 2000 captcha images consisting of numerical characters only
> 
> Caveats:
> 
> 1. The statistics printed at the end of the script are off right now, does not account for overwriting existing captcha images. This shouldn't really matter though, what is important is that they show that each character occurs more than once and on average, similar amounts of time.
> 
> image

With respect to your logging at the end, shouldn't this file in theory generate the entire dataset? Overwritten data samples should be irrelevant no?

File: dataset_generator.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        index = random.randint(0, len(characters) - 1)
+        captcha_elements.append(characters[index])
+        
+        # Update frequency for dataset statistics
+        old_frequency = frequency.get(characters[index])
+        frequency.update({ characters[index]: old_frequency + 1 })
+    
+    # Create actual captcha image
+    text = "".join(captcha_elements)
+    generator.write(text, f"./dataset/{text}.png")
```
Comment (natanleung) [inline dataset_generator.py:29]: I assume this will recursively create `./dataset` because it does not yet exist

**Comment by fcpranav** [comment]
Is it empty? Shouldn't be

File: dataset_generator.py (hunk 1 above)
Comment (fcpranav) [reply dataset_generator.py:29]: Sadly no, you need to make sure one exists, but that's a minor caveat that is just a slight inconvenience for a script we will barely use.

**Comment by natanleung** [comment]
> Is it empty? Shouldn't be. About first comment, yah, overwritten data is irrelevant. It is just that your end dataset will have a few less images than the count.

Nah its cool. Github tripping.

**Comment by natanleung** [comment]
Approved.

**Pull request merged**